### PR TITLE
Finished header

### PR DESCRIPTION
**Pull request by srallen**
For #5

This finishes the header as designed by Becky with all of the zooniverse.org header links except Admin and News. I didn't include the Admin link because I need to think about how I want to integrate that first. News isn't include because last chat with Becky about it is that she wants to make it a standard nav link to a news page rather than a drop down menu. 

Notifications and Messages links are now present when signed in, but are just regular links to those pages and do not include the websocket functionality as on the main zooniverse.org

There is now a sign in modal that has a button to link to the standard panoptes oauth devise page. There is also a button now that directly links to the panoptes oauth google sign in (google is quite picky about [how this looks](https://developers.google.com/identity/branding-guidelines).

I removed the placeholder about page from zoo-reduxify.

TO DO:
- [x] Use the local storage like I setup with the join page to redirect users back to where they were when they clicked to sign in. 
- [x] A sub-header nav for the Edu API site itself
- Extract the main zooniverse header into ZRC repo. This I'll do separate from this PR.

### CODE REVIEW

**Comment by srallen** [comment]
I rebased this branch against the programs branch because there were some routes fixes in that branch that I wanted here. So, #52 will need to merge first before this is merged. 

I reworked the redirect using local storage from the join page functionality to be more general purpose. Now if someone logs in wherever on the app, they'll get redirected back to where they were.

For testing purposes, the sign in with google has a known issue on staging (zooniverse/Panoptes#2024), so you'll have to use production to test that.

**Comment by srallen** [comment]
Just one more note, I haven't checked with @beckyrother about the app header, but a style pass can happen after this functional PR.

**Comment by srallen** [comment]
The redirect isn't working quite as expected, so I'm looking into it....

**Comment by srallen** [comment]
OK now this should be working. Just an FYI, the `location.pathname` property provided by history.js when using a hash router isn't the same as the window object's `location.pathname`. 🤦‍♀️

File: src/components/layout/ZooHeader.jsx [hunk 1]
```diff
@@ -23,8 +24,14 @@ const ZooHeader = (props) => {
 
 ZooHeader.defaultProps = {
   authContainer: null,
-  logoHomeLink: <ZooniverseLogo height="1.25em" width="1.25em" />,
-  mainHeaderNavList: null
+  logoHomeLink: <Anchor className="site-header__link" href="http://www.zooniverse.org"><ZooniverseLogo height="1.25em" width="1.25em" /></Anchor>,
+  mainHeaderNavList: [
+    <Anchor className="site-header__link--small" href="http://www.zooniverse.org/projects" label="Projects" />,
+    <Anchor className="site-header__link--small" href="http://www.zooniverse.org/about" label="About" />,
+    <Anchor className="site-header__link--small" href="http://www.zooniverse.org/about" label="About" />,
```
Comment (shaunanoordin) [inline src/components/layout/ZooHeader.jsx]: We have two About links?

File: src/containers/common/ProgramHomeContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      Actions.getProgram({ programs: this.props.programs, param: this.props.match.params.program });
+    }
+  }
+
+  componentWillUnmount() {
+    Actions.programs.selectProgram(PROGRAMS_INITIAL_STATE.program);
+  }
+
+  render() {
+    // How do I pass props down consistently? They only get initial props, not the updated ones.
```
Comment (shaunanoordin) [inline src/containers/common/ProgramHomeContainer.jsx]: What were you trying to pass down, but didn't get through/the update didn't register? The props.selectedProgram?

**Comment by shaunanoordin** [comment]
I've written up the review for this PR _(in short: LGTM, minor fixes like the double About links, mostly awaiting #52 to go through)_ but I've decided against posting it _yet_ since I might repeat a bunch of things from my upcoming PR #52 review. In retrospect, I should've reviewed that first, so I'll do so right now.

File: src/components/layout/ZooHeader.jsx (hunk 1 above)
Comment (srallen) [reply src/components/layout/ZooHeader.jsx]: One of those should've been the link Get Involved. Oops!

**Comment by srallen** [comment]
This is now rebased and the Get Involved link is fixed.

**Comment by srallen** [comment]
@shaunanoordin I'll look over this again tomorrow AM. Google sign in does have a known issue on staging, but it should work on production. It sounds like something might be going wrong there, so I'll double check that.

**Comment by srallen** [comment]
@shaunanoordin I fixed the regression with the join page redirecting back to the program home. In addition: 
- A minor style tweak to the header (added the border bottom separator)
- Fixed bug with the conditional to show placeholder redirect message on the join page UI. This placeholder is there for feature parity with the Wildcam redirect that goes to the map viewer like on the current Wildcam lab.
- I fixed the incorrect status prop on the success join page UI
- I added a conditional to hide the app header on both the home page and on the join page. I've hidden it on the join page so I2A students do not find the teacher materials. 

The google sign in will take a bit more time to debug. In addition to the known issue with staging, I think you're right that the wrong environment is being used when attempting to test in production. I'd like to try to solve this separately from this PR.

**Comment by shaunanoordin** [comment]
## PR Review (Update)
* This update PR now properly allows users to follow a Join link.
* The Google Sign In issue will be investigated separately.

screen shot 2017-09-28 at 16 17 50
_zootester2 viewing zootester1's Classroom Join Link._

screen shot 2017-09-28 at 16 19 13
_zootester2 successfully joining a Classroom. Note that at this moment, we don't have a way for Students to view the list of her/his Classrooms (AFAIK)_

screen shot 2017-09-28 at 16 19 33
_zootester2 sees this message when she attempts to join a Classroom that she's already a part of. Or an invalid link. Either way._

### Status
LGTM! This PR is approved and will be merged. Another Issue will be opened in regards to investigating the Google Sign In matter.

**Pull request merged**